aima-core/src/main/java/aima/core/logic/planning/PlanningProblemFactory.java

```diff
@@ -0,0 +1,56 @@
+package aima.core.logic.planning;
+
+import aima.core.logic.fol.parsing.ast.Constant;
+import aima.core.logic.fol.parsing.ast.Variable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class PlanningProblemFactory {
+    public static Problem airCargoTransportProblem() {
+        State initialState = new State("At(C1,SFO)^At(C2,JFK)^At(P1,SFO)" +
+                "^At(P2,JFK)^Cargo(C1)^Cargo(C2)^Plane(P1)^Plane(P2)^Airport(JFK)^Airport(SFO)");
+        State goalState = new State("At(C1,JFK)^At(C2,SFO)");
+        Variable c = new Variable("c");
+        Variable p = new Variable("p");
+        Variable a = new Variable("a");
+        Variable from = new Variable("from");
+        Variable to = new Variable("to");
+        ArrayList variables = new ArrayList<>(Arrays.asList(c, p, a));
+        ArrayList flyVars = new ArrayList<>(Arrays.asList(p, from, to));
+        ActionSchema loadAction = new ActionSchema("Load", variables,
```

Comment:
This is good as is, but maybe it would be easier if ActionSchema took a string, not a ArrayList, as the second argument? Here the string would be "c, p, a", and we wouldn't need the 7 declarations of variables and ArrayLists.

Reply:
Yeah, strings are a lot better. I will improve this in the next PR.